packages/protocol/src/action-protocol/clipboard.ts

```diff
@@ -14,80 +14,154 @@
 
-import { isObject, isString } from '../utils/typeguard-util';
-import { generateRequestId, isActionKind, Operation, RequestAction, ResponseAction } from './base-protocol';
+import { hasObjectProp } from '../utils/type-util';
+import { Action, Operation, RequestAction, ResponseAction } from './base-protocol';
 import { EditorContext } from './types';
 
 /**
  * Requests the clipboard data for the current editor context, i.e., the selected elements, in a clipboard-compatible format.
+ * The corresponding namespace declares the action kind as constant and offers helper functions for type guard checks
+ * and creating new `RequestClipboardDataActions`.
  */
-export class RequestClipboardDataAction implements RequestAction<SetClipboardDataAction> {
-    static readonly KIND = 'requestClipboardData';
+export interface RequestClipboardDataAction extends RequestAction<SetClipboardDataAction> {
+    /**
+     * The unique action kind.
+     */
+    kind: typeof RequestClipboardDataAction.KIND;
 
-    constructor(
-        public readonly editorContext: EditorContext,
-        public readonly requestId: string = generateRequestId(),
-        public readonly kind: string = RequestClipboardDataAction.KIND
-    ) {}
+    /**
+     * The current editor context.
+     */
+    editorContext: EditorContext;
+}
+
+export namespace RequestClipboardDataAction {
+    export const KIND = 'requestClipboardData';
 
-    static create(editorContext: EditorContext): RequestAction<SetClipboardDataAction> {
-        return new RequestClipboardDataAction(editorContext);
+    export function is(object: any): object is RequestClipboardDataAction {
+        return RequestAction.hasKind(object, KIND) && hasObjectProp(object, 'editorContext');
     }
-}
 
-export function isRequestClipboardDataAction(action: any): action is RequestClipboardDataAction {
-    return isActionKind(action, RequestClipboardDataAction.KIND) && isObject(action, 'editorContext') && isString(action, 'requestId');
+    export function create(editorContext: EditorContext, options: { requestId?: string } = {}): RequestClipboardDataAction {
+        return {
+            kind: KIND,
+            requestId: '',
+            editorContext,
+            ...options
+        };
+    }
 }
 
 /**
  * Server response to a {@link RequestClipboardDataAction} containing the selected elements as clipboard-compatible format.
+ * The corresponding namespace declares the action kind as constant and offers helper functions for type guard checks
+ * and creating new `SetClipboardDataActions`.
  */
-export class SetClipboardDataAction implements ResponseAction {
-    static readonly KIND = 'setClipboardData';
-
-    constructor(
-        public readonly clipboardData: ClipboardData,
-        public readonly responseId: string = '',
-        public readonly kind: string = SetClipboardDataAction.KIND
-    ) {}
+export interface SetClipboardDataAction extends ResponseAction {
+    /**
+     * The unique action kind.
+     */
```

Comment:
I'd spare such comments, also if it hurts to be "incomplete". But imho they just add noise and add no information.